### ISSUE TITLE
UserWarning: file not found

### ISSUE THREAD

**Issue by grantjenks** [grantjenks/python-diskcache#3]
Occurred today in Travis at 1.6.3 on both CPython 3.4 and PyPy.

```
======================================================================
ERROR: Stress test multiple threads and processes.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/travis/virtualenv/pypy-2.5.0/site-packages/nose/case.py", line 197, in runTest
    self.test(*self.arg)
  File "/home/travis/build/grantjenks/python-diskcache/tests/stress_test_fanout.py", line 280, in stress_test_mp
    stress_test()
  File "/home/travis/build/grantjenks/python-diskcache/tests/stress_test_fanout.py", line 244, in stress_test
    cache.check()
  File "/home/travis/build/grantjenks/python-diskcache/diskcache/fanout.py", line 135, in check
    return sum((shard.check(fix=fix) for shard in self._shards), [])
  File "/home/travis/build/grantjenks/python-diskcache/diskcache/fanout.py", line 135, in <genexpr>
    return sum((shard.check(fix=fix) for shard in self._shards), [])
  File "/home/travis/build/grantjenks/python-diskcache/diskcache/core.py", line 873, in check
    warnings.warn('file not found: %s' % full_path)
UserWarning: file not found: tmp/000/90/06/d93b748a5ebea1fee3fc3fe441bf.val
```

This is worrisome. I don't yet understand how a file was lost.

**Comment by grantjenks** [comment]
Happened again:
```
======================================================================
ERROR: Stress test multiple threads and processes.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/travis/virtualenv/python3.5.0/lib/python3.5/site-packages/nose/case.py", line 198, in runTest
    self.test(*self.arg)
  File "/home/travis/build/grantjenks/python-diskcache/tests/stress_test_fanout.py", line 280, in stress_test_mp
    stress_test()
  File "/home/travis/build/grantjenks/python-diskcache/tests/stress_test_fanout.py", line 244, in stress_test
    cache.check()
  File "/home/travis/build/grantjenks/python-diskcache/diskcache/fanout.py", line 135, in check
    return sum((shard.check(fix=fix) for shard in self._shards), [])
  File "/home/travis/build/grantjenks/python-diskcache/diskcache/fanout.py", line 135, in <genexpr>
    return sum((shard.check(fix=fix) for shard in self._shards), [])
  File "/home/travis/build/grantjenks/python-diskcache/diskcache/core.py", line 873, in check
    warnings.warn('file not found: %s' % full_path)
UserWarning: file not found: tmp/000/ca/fe/18b02980a82f7ce32a8b559c280f.val
```

**Comment by grantjenks** [comment]
I believe the use of transactions will eliminate this class of error.

Fixed and deployed in v2.0.2 on PyPI.
